### PR TITLE
MODDICONV-281 - Enable ability to edit Job profile: Default - Create SRS MARC Authority

### PR DESCRIPTION
**Pull request by sherzodn**
## Purpose
_The "400" error appears when user tries to save edited "Default - Create SRS MARC Authority" job profile with the one of the following updated fields:
a) "Name*" or "Accepted data type*" or "Description";
b) "Name*" and "Accepted data type*" and "Description";_

## Approach
_ DEFAULT_CREATE_SRS_MARC_AUTHORITY_JOB_PROFILE_ID should be skipped during update request_

#### TODOS and Open Questions
- [ ] Use GitHub checklists. When solved, check the box and explain the answer.

## Learning
https://issues.folio.org/browse/MODDICONV-281 
and more detailed here https://issues.folio.org/browse/UIDATIMP-1318

### CODE REVIEW

File: mod-data-import-converter-storage-server/src/main/java/org/folio/rest/impl/DataImportProfilesImpl.java [hunk 1]
```diff
@@ -230,7 +228,7 @@ public void deleteDataImportProfilesJobProfilesById(String id, String lang, Map<
                                                       Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
     vertxContext.runOnContext(v -> {
       try {
-        if (canDeleteOrUpdateProfile(id, JOB_PROFILES)) {
```
Comment (psmagin) [inline mod-data-import-converter-storage-server/src/main/java/org/folio/rest/impl/DataImportProfilesImpl.java:233]: Why did you change the delete endpoint? The issue is about updating...

**Review by psmagin — CHANGES_REQUESTED** [review]

File: mod-data-import-converter-storage-server/src/main/java/org/folio/rest/impl/DataImportProfilesImpl.java (hunk 1 above)
Comment (sherzodn) [reply mod-data-import-converter-storage-server/src/main/java/org/folio/rest/impl/DataImportProfilesImpl.java:233]: Yes, I removed authority job profile id from JOB_PROFILES array. We are able to update it, but not delete. So that, I had to  refactor the logic in delete API

**Review by psmagin — CHANGES_REQUESTED** [review]
Please make sure that all methods have proper naming and not conflict with each other. Ex: canDeleteOrUpdateProfile and canNotBeDeleted.

Using not in method name is not a good practice, better invert condition in a place of method call.

**Review by psmagin — APPROVED** [review]
